src/Nox.Solution/NoxSolutionBuilder.cs

```diff
@@ -12,17 +13,18 @@ public class NoxSolutionBuilder
         private const string DesignFolderBestPractice = "Best practice is to create a '.nox' folder in your solution folder and in there a 'design' folder which contains your <solution-name>.solution.nox.yaml";
 
         private string _yamlFilePath = string.Empty;
-        private IMacroParser[]? _macroParsers;
+
+        private IMacroParser _environmentVariableParser = new EnvironmentVariableMacroParser(new EnvironmentProvider());
 
         public NoxSolutionBuilder UseYamlFile(string yamlFilePath)
         {
             _yamlFilePath = Path.GetFullPath(yamlFilePath);
             return this;
         }
 
-        public NoxSolutionBuilder UseMacros(IMacroParser[] macroParsers)
+        public NoxSolutionBuilder UseEnvironmentMacroParser(EnvironmentVariableMacroParser parser)
```

Comment:
Should we allow the client to setup any `IMacroParser` instead of the concrete `EnvironmentVariableMacroParser`, because this way the user can not extend the behavior? If so we could also update the name to `UseMacroParser`?

Reply:
I suggest we do a seperate builder class and limit the parsers. Most people will never want to specify and override parsers (mainly for tests I suspect). The "Canonical Nox Yaml Format for Solutions" will always support a strict syntax and macro variables should be handled by default (if you agree)

Reply:
I'll fix